rs/nns/governance/src/neuron_store/metrics.rs

```diff
@@ -98,6 +100,39 @@ impl NeuronMetrics {
         self.public_neuron_subset_metrics
             .increment(voting_power_economics, now_seconds, neuron);
     }
+
+    /// This could modify either declining_voting_power_neuron_subset_metrics, or
+    /// fully_lost_voting_power_neuron_subset_metrics (but not both), since
+    /// those categories are mutually exclusive.
+    fn increment_declining_voting_power_or_fully_lost_voting_power_neuron_subset_metrics(
+        &mut self,
+        voting_power_economics: &VotingPowerEconomics,
+        now_seconds: u64,
+        neuron: &Neuron,
+    ) {
+        let seconds_since_voting_power_refreshed =
+            // Here, we assume that the neuron was not refreshed in the future.
+            // This doesn't always hold in tests though, due to the difficulty
+            // of constructing realistic data/scenarios.
+            now_seconds.saturating_sub(neuron.voting_power_refreshed_timestamp_seconds());
+        let Some(seconds_losing_voting_power) = seconds_since_voting_power_refreshed
+            .checked_sub(voting_power_economics.get_start_reducing_voting_power_after_seconds())
```

Comment:
This is kind of a clever way to do this, but wouldn't it be simpler to use a comparison?

Reply:
Will do.

Is this really exotic??

Yes, `None` is not as explicit as `observed < limit`.

For better or worse, we have pledged to Andy to use checked_sub (et. al.) instead of the `-` operator. Therefore, I assume the team is familiar with checked_sub. In that case, I think this should not be considered particularly "clever".

# Tangent

In an ideal world, we wouldn't use u64 et. al. Instead, we should be like Motoko and Python (and to a lesser degree, the Google C++ style guide): Int. Int is closed under subtraction (and all the other math operations, except divide by 0 is ofc undefined), just like math. In that case, I would consider doing it like this:

```
let excess = observed - limit;
if excess < 0 {
  return;
}

if excess < uber_limit {
  // merely stale
} else {
  // expired
}
```